app/javascript/providers/SearchSelectionProvider.jsx

```diff
@@ -70,10 +70,36 @@ export default function SearchSelectionProvider(props) {
     }
     setSelection(newSelection)
   }
+  /**
+   * if there are lowercased accessions in the search terms,
+   * make them uppercased before performing the server search
+   */
+  function uppercaseAccessionTerms(selection) {
```

Comment:
Thinking out loud - why did you opt for doing the conversion client-side as opposed to server-side?  The approach you adopted works perfectly well, but if someone makes a direct search request through the API, then the conversion will not take place.  Granted, this is likely exceedingly rare, but it could happen.

After looking through the code, it seems we could have the same effect with very minimal changes if we updated the `StudyAccession.sanitize_accessions` method (this is what does the matching in search requests):

```
# line 11 in study_accession.rb
ACCESSION_SANITIZER = /[^SCP\d+$]/i
...

# line 37 in study_accession.rb
accession_string = term.strip.gsub(ACCESSION_SANITIZER, '').upcase
```

This generates the following output:
```
irb(main):001:0> terms = %w[foo bar scp1 SCP2 scp scpblah]
=> ["foo", "bar", "scp1", "SCP2", "scp", "scpblah"]
irb(main):002:0> StudyAccession.sanitize_accessions(terms)
=> ["SCP1", "SCP2"]

```